pkg/cli/codearea.go

```diff
@@ -196,6 +200,71 @@ func (w *codeArea) handlePasteSetting(start bool) bool {
 	return true
 }
 
+// Try to expand a simple abbreviation that ignores word boundaries.
+func (w *codeArea) expandSimpleAbbr() {
+	var abbr, full string
+	// Find the longest matching abbreviation.
+	w.Abbreviations(func(a, f string) {
+		if strings.HasSuffix(w.inserts, a) && len(a) > len(abbr) {
+			abbr, full = a, f
+		}
+	})
+	if len(abbr) > 0 {
+		c := &w.State.Buffer
+		*c = CodeBuffer{
+			Content: c.Content[:c.Dot-len(abbr)] + full + c.Content[c.Dot:],
+			Dot:     c.Dot - len(abbr) + len(full),
+		}
+		w.resetInserts()
+	}
+}
+
+// Try to expand a small word abbreviation.
+func (w *codeArea) expandSmallWordAbbr(triggerRune rune, triggerLen int) {
+	// We only do small word expansions at the end of the line and only if
+	// there are characters in the insert buffer other than the trigger rune.
+	c := &w.State.Buffer
+	if len(w.inserts) <= triggerLen || c.Dot < len(c.Content) {
+		return
+	}
+
+	inserts := w.inserts[:len(w.inserts)-triggerLen] // ignore abbr trigger rune
+	var abbr, full string
+	// Find the longest matching abbreviation.
+	w.SmallWordAbbreviations(func(a, f string) {
+		if len(a) >= len(w.inserts) || len(a) < len(abbr) {
```

Comment:
The check `len(a) >= len(w.inserts)` is unnecessary here as its case will be caught by `strings.HasSuffix(inserts, a)` below.

Reply:
Yes, this is arguably an instance of premature optimization but given the other, necessary, length check it seemed reasonable to do both at the same time to make it clear to the reader we can't possibly match the abbreviation if the insert buffer is too short and, potentially, shave a few nanoseconds off the run time for the common case.